subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/ClientProvidedBuildActionRunner.java

```diff
@@ -44,22 +43,14 @@ public void run(BuildAction action, final BuildController buildController) {
         }
 
         final GradleInternal gradle = buildController.getGradle();
+        gradle.getStartParameter().setConfigureOnDemand(false);
```

Comment:
❓ What's the connection between `ClientProvidedBuildAction` and `gradle.getStartParameter().setConfigureOnDemand(false)`?